scrapy/http/request/form.py

```diff
@@ -48,7 +49,11 @@ def from_response(cls, response, formname=None, formid=None, formnumber=0, formd
         form = _get_form(response, formname, formid, formnumber, formxpath)
         formdata = _get_inputs(form, formdata, dont_click, clickdata, response)
         url = _get_form_url(form, kwargs.pop('url', None))
-        method = kwargs.pop('method', form.method)
+
+        method = kwargs.pop('method', form.method).upper()
+        if method not in cls.valid_form_methods:
+            raise ValueError('Invalid form method in chosen form')
```

Comment:
I wonder if we should follow standard more closely, and consider unknown and invalid methods as GET (see https://www.w3.org/TR/html5/sec-forms.html#element-attrdef-form-method):

> The invalid value default for these attributes is the GET state. The missing value default for the method attribute is also the GET state. (There is no missing value default for the formmethod attribute.)

Reply:
Yep, I agree with that, I'm going to modify the PR.